src/core/demangle.d

```diff
@@ -207,8 +207,9 @@ pure @safe:
         if( val.length )
         {
             assert( contains( dst[0 .. len], val ) );
+            if (len + val.length > dst.length)
+                overflow();
             debug(info) printf( "removing (%.*s)\n", cast(int) val.length, val.ptr );
-
             size_t v = &val[0] - &dst[0];
             for (size_t p = v; p < len; p++)
```

Comment:
This loop is the culprit: `len` is wrong, it should be `len-val.length`, i.e. the change to len below could just be moved up.

Reply:
done